core/common/src/main/java/alluxio/network/protocol/databuffer/DataNettyBuffer.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.network.protocol.databuffer;
+
+import com.google.common.base.Preconditions;
+import io.netty.buffer.ByteBuf;
+
+import java.nio.ByteBuffer;
+
+/**
+ * A DataBuffer with the underlying data being a {@link ByteBuf}.
+ */
+public final class DataNettyBuffer implements DataBuffer {
```

Comment:
This may be an artifact of bad naming for previous classes, but would `ByteBufDataBuffer` be more relevant?

Reply:
Given that we also have `DataBuffer` based on nio `ByteBuffer`, how about renaming it to `NioDataBuffer` and `NettyDataBuffer` to avoid confusion between the name `ByteBuf` and `ByteBuffer`?

Reply:
Updated.